addons/l10n_es_pos/views/pos_order_views.xml

```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" ?>
+<odoo>
+    <!-- POS order form -->
+    <record id="view_pos_pos_form_simplified_invoice" model="ir.ui.view">
+        <field name="name">pos.order.form</field>
+        <field name="model">pos.order</field>
+        <field name="inherit_id" ref="point_of_sale.view_pos_pos_form" />
+        <field name="arch" type="xml">
+            <field name="name" position="after">
+                <field name="l10n_es_unique_id" readonly="1" invisible="not l10n_es_unique_id" />
+            </field>
+        </field>
+    </record>
+    <!-- POS order tree -->
+    <record id="view_pos_order_tree" model="ir.ui.view">
+        <field name="model">pos.order</field>
+        <field name="inherit_id" ref="point_of_sale.view_pos_order_tree" />
+        <field name="arch" type="xml">
+            <field name="pos_reference" position="before">
+                <field name="pos_is_spanish" column_invisible="True"/>
+                <field name="l10n_es_unique_id" invisible="not pos_is_spanish" />
```

Comment:
can't you do `invisible="not config_id.is_spanish"` (make sure `config_id` is present in the view)?
And with that, maybe we can get rid of the related field in the `pos.order` model.